apps/backend/src/interfaces/soroban/index.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  // Account,
+  // Address,
+  Contract,
+  // FeeBumpTransaction,
+  // hash,
+  Keypair,
+  // Operation,
+  rpc,
+  // StrKey,
+  // Transaction,
+  TransactionBuilder,
+  // xdr,
+} from '@stellar/stellar-sdk'
+
+import { SingletonBase } from 'api/core/framework/singleton/interface'
+import { logger } from 'config/logger'
+import { STELLAR } from 'config/stellar'
+
+import { ERRORS } from './helpers/errors'
+import { ISorobanService, SimulateContract, SimulationResult } from './types'
+
+export default class Soroban extends SingletonBase implements ISorobanService {
+  public rpcClient: rpc.Server
+  public networkPassphrase: string
+  public timeoutInSeconds: number
+  public fee: string
+  public sourceAccountKP: Keypair
+
+  constructor() {
+    super()
+    this.rpcClient = new rpc.Server(STELLAR.SOROBAN_RPC_URL, {
+      allowHttp: STELLAR.SOROBAN_RPC_URL.startsWith('http://'),
+    })
+    this.networkPassphrase = STELLAR.NETWORK_PASSPHRASE
+    this.timeoutInSeconds = 60
+    this.fee = STELLAR.MAX_FEE
+    this.sourceAccountKP = Keypair.fromSecret(STELLAR.SOURCE_ACCOUNT.PRIVATE_KEY) // TODO: get account KP data with passkey
+  }
+
+  /**
+   * Simulates a Soroban contract method.
+   * @param contractId - The ID of the Soroban contract.
+   * @param method - The method to call.
+   * @param args - The arguments for the contract call.
+   * @param signers - (Optional) Signers for the contract call.
+   * @returns The transaction and simulation result.
+   */
+  public async simulateContract({
```

Comment:
```suggestion
  public async simulateContractMethod({
```

Wdyt? Looks more explicit about what it is